src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -689,67 +703,70 @@ private static unsafe ProcessInfo[] GetProcessInfos(IntPtr dataPtr)
                 IntPtr currentPtr = (IntPtr)((long)dataPtr + totalOffset);
                 ref SystemProcessInformation pi = ref *(SystemProcessInformation *)(currentPtr);
 
-                // get information for a process
-                ProcessInfo processInfo = new ProcessInfo();
                 // Process ID shouldn't overflow. OS API GetCurrentProcessID returns DWORD.
-                processInfo.ProcessId = pi.UniqueProcessId.ToInt32();
-                processInfo.SessionId = (int)pi.SessionId;
-                processInfo.PoolPagedBytes = (long)pi.QuotaPagedPoolUsage; ;
-                processInfo.PoolNonPagedBytes = (long)pi.QuotaNonPagedPoolUsage;
-                processInfo.VirtualBytes = (long)pi.VirtualSize;
-                processInfo.VirtualBytesPeak = (long)pi.PeakVirtualSize;
-                processInfo.WorkingSetPeak = (long)pi.PeakWorkingSetSize;
-                processInfo.WorkingSet = (long)pi.WorkingSetSize;
-                processInfo.PageFileBytesPeak = (long)pi.PeakPagefileUsage;
-                processInfo.PageFileBytes = (long)pi.PagefileUsage;
-                processInfo.PrivateBytes = (long)pi.PrivatePageCount;
-                processInfo.BasePriority = pi.BasePriority;
-                processInfo.HandleCount = (int)pi.HandleCount;
-
-
-                if (pi.ImageName.Buffer == IntPtr.Zero)
+                var processInfoProcessId = pi.UniqueProcessId.ToInt32();
```

Comment:
you could assert this is not -1 if you liek.

Reply:
done